src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.PackageDependencyResolution.targets

```diff
@@ -414,8 +414,12 @@ Copyright (c) .NET Foundation. All rights reserved.
       <Reference Include="@(ResolvedFrameworkAssemblies)" />
     </ItemGroup>
     
+    <!-- If there are any references from a NuGet package that match a simple reference which
```

Comment:
Then conflict resolution will later decide if the simple name wins or the package file does?  Where does the path to the package file get preserved?

Reply:
The `ResolvedCompileFileDefinitions` items have the `HintPath` set.  This join basically changes the itemspec from the full path to the DLL in the NuGet package to the simple name of the assembly.  That's for the benefit of the project system so it understands that the simple reference didn't just disappear.  Later on in conflict resolution, if the platform copy of the DLL wins, then these items will be removed from the result, and a new simple name item will be added.